### DISCUSSION TITLE
Using local volumes

### DISCUSSION THREAD

**Discussion by muddywellyboots** [biigle/core#377] [Q&A] [answered]
Hi,
I've installed the GPU branch of BIIGLE on Ubuntu as described in the instructions.

I'm struggling to use files from a local directory. If I wanted to use files in say '/home/john/Pictures' , would it be 'local:///home/john/Pictures' I use? Or does the path depend on where BIIGLE is installed?

p.s Thanks for writing some very useful software!

Many thanks.

**Comment by muddywellyboots** [comment]
I've also tried a simple http server locally with "python3 -m http.server' in a folder with some images, then trying "http://0.0.0.0:8000/" as the Volume url, but I get a connection error.

Thanks

**Reply by mzur** [reply]
The Docker containers in which BIIGLE is running can't access your host machine in that way (just as they can't access the host file system). It's a little complicated to find out the correct IP address of the host machine to use here. Maybe my other answer (regarding the file system) is already sufficient. If not, we can discuss this further.

**Comment by mzur** [comment, marked as answer]
As BIIGLE runs inside Docker containers, the file system of your host machine is not accessible by default. File system volumes must be explicitly configured with Docker for BIIGLE to be able to access files on your host machine. These volumes are configured in the file [`docker-compose.yaml`](https://github.com/biigle/biigle/blob/0fe99db7b3f6363f60a8f8bc679adfebd32ac104/docker-compose.yaml#L14). In the default configuration, only the `./storage` directory (of the biigle repository) is mounted as file system volume. You can either store all images/videos that you want available in BIIGLE there (in `./storage/images`, see below) or add new file system volumes. Example:
```yaml
- /home/john/Pictures:/vol/Pictures
```
The syntax of the volume definition is `[source path of host]:[target path of container]`.

If you add new file system volumes, you also have to configure them as new storage disks in BIIGLE. This is done in the file [`./build/config/filesystems.php`](https://github.com/biigle/biigle/blob/0fe99db7b3f6363f60a8f8bc679adfebd32ac104/build/config/filesystems.php#L47-L50). In the default configuration, BIIGLE defines the directory `./storage/images` as the `local` storage disk. So if you have a directory of images `./storage/images/MyImages`, for example, you can create a new volume in BIIGLE with the URL `local://MyImages`. You can also add new storage disks. Example (following the example above):
```php
'pictures' => [ // This is the name of the storage disk.
   'driver' => 'local',
   'root' => '/vol/Pictures', // This is the target path defined above.
],
```
If you now have the directory of images `/home/john/Pictures/MyImages`, you can create a new volume in BIIGLE with the URL `pictures://MyImages`. You need to rebuild the Docker containers and restart BIIGLE whenever you change anything in `filesystems.php`.

**Reply by muddywellyboots** [reply]
Hi @mzur,
That worked perfectly. Thank you for your advice.
